eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -59,47 +60,47 @@
     public Resource getCoreInfo(@PathVariable("variantCoreString") String variantCoreString,
                                 @RequestParam(name = "species") String species,
                                 @RequestParam(name = "assembly") String assembly,
-                                HttpServletResponse response) throws IllegalArgumentException{
+                                HttpServletResponse response) throws IllegalArgumentException {
         initializeQuery();
         try {
             checkParameters(variantCoreString, null, null, species, assembly);
         } catch (IllegalArgumentException e) {
-            System.out.println(e.getMessage());
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
             return new Resource<>(setErrorQueryResponse(e.getMessage()));
         }
 
         MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
 
-        List<VariantWithSamplesAndAnnotation> variantEntities;
-        Long numTotalResults;
+        Optional<VariantWithSamplesAndAnnotation> variantEntity;
 
         try {
-            variantEntities = getVariantEntitiesByParams(variantCoreString, null, null);
-            numTotalResults = (long) variantEntities.size();
-            if (variantEntities.size() == 0) {
-                response.setStatus(HttpServletResponse.SC_NOT_FOUND);
-                return new Resource<>(setQueryResponse(variantEntities));
-            }
+            variantEntity = getVariantByCoordinatesAndAnnotationVersion(variantCoreString, null, null);
         } catch (AnnotationMetadataNotFoundException ex) {
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
             return new Resource<>(setQueryResponse(ex.getMessage()));
+        } catch (IllegalArgumentException ex) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setQueryResponse(ex.getMessage()));
         }
 
         List<Variant> variantList = new ArrayList<>();
-        variantEntities.forEach(variantEntity->{
-            Variant variant = new Variant(variantEntity.getChromosome(), variantEntity.getStart(),
-                    variantEntity.getEnd(), variantEntity.getReference(),variantEntity.getAlternate());
-            variant.setIds(variantEntity.getIds());
-            variantList.add(variant);
-        });
+        if (!variantEntity.isPresent()) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return new Resource<>(setQueryResponse(variantList));
+        }
+
+        Variant variant = new Variant(variantEntity.get().getChromosome(), variantEntity.get().getStart(),
```

Comment:
In order to avoid many redundant calls, extract the result from `variantEntity.get()` to a local variable, which could be called something like `retrievedVariant`.

Reply:
Fixed :)